docs/design/backups.md

```diff
@@ -0,0 +1,105 @@
+Once this is complete, an event will notify completion, we'll also update the resource state to `Complete`.
+If an error occured during the backup process, this will also be shown by setting the resource status to `Failed`.
+
+## Backup Schedule Resource
+
+We would like to be able to take automated backups on a given schedule. 
+A second custom resource will be added to support this: 
+
+```yaml
+apiVersion: etcd.improbable.io/v1alpha1
+kind: EtcdBackupSchedule
+metadata:
+  name: my-backup-schedule
+  namespace: default
+spec:
+  schedule: "*/1 * * * *"
+  successfulBackupsHistoryLimit: 30
+  failedBackupsHistoryLimit: 5
+  backupTemplate:
+    type: snapshot
+    destination:
+      gcsBucket:
+        bucketName: my-cluster-backup-bucket
+        creds:
+          secretRef: 
+            name: my-cluster-backup-bucket-credentials
+            namespace: default
+```
+
+This resource appears very similar to the `EtcdBackup` resource, with the addition of a `schedule` field:
+This field allows backups to be taken on a crontab-notation schedule, behaving similarly to the Kubernetes [CronJob](https://kubernetes.io/docs/tasks/job/automated-tasks-with-cron-jobs/#schedule) resource.
+
+The controller will maintain a pool of go routines, one for each backup schedule resource.
+These go routines will simply sleep until the crontab fires, at which point they will create a labelled `EtcdBackup` resource from the given configuration.
+The controller reconcile process will make sure that the pool of goroutines are a reflection of the desired configuration.
+New routines will be created when a resource is applied, routines will be recreated when a resource is changed, and deleted when a resource is removed.
+
+Once a backup has finished, the controller will ensure that we are only keeping the specified number of `EtcdBackup` resources around, and will clean up any old resources respecting the `successfulBackupsHistoryLimit` and `failedBackupsHistoryLimit` fields.
+
+## Backup Restoration
+
+Providing tooling to restore backups is out of scope of this design.
+We have judged it to be sufficient to provide documentation on how to restore backups, but not provide any built-in mechanisms to perform the restoration for you.
+The ETCD tooling does well at restoring the backup snapshots and we do not feel like there is much value to be added there.
+
+## Alternatives
+
+The [Velero](https://github.com/vmware-tanzu/velero) project aims to provide tools to back up all important data in a Kubernetes cluster. 
+We considered building an integration in to this project to do the ETCD backups for us - and there was no technical reason we found why this wouldn't be possible. 
+However we would like to offer something out of the box for users who do not want to run Velero, so chose to build this in to the operator as a start.
+
```

Comment:
it'll need to be a bit more complex than this.. we'd need to
1. install etcdctl/kubectl/gsutil or load from a prebuilt container
2. take the backup, as written
3. load the cloud storage (GCP) service account credentials from the k8s API using the kubernetes service account token mounted into the pod, making sure RBAC is configured for this.
4. call in to gsutil, supplying it with the credentials we just read, to push the snapshot to remote storage

its getting to the point where a script doing this would be pretty difficult to maintain - and that's before we think about supporting different storage mechanisms and/or different backup strategies

Reply:
I'd almost consider if we use `Job`/`CronJob` an implementation detail. I think a user is going to find it easier to configure a custom `EtcdBackup` or `EtcdBackupSchedule` resource rather than fiddling about with pod templates in a `CronJob`. Even if that `EtcdBackupSchedule` just makes a `CronJob` to do the dirty work.

For now I'm happy with the controller just doing things, but I could totally see us replacing that with 'the controller makes a Job or CronJob' in the future.